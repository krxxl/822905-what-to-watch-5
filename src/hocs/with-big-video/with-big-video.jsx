import React, {PureComponent} from 'react';
import PropTypes from 'prop-types';
import {getFilmById} from '../../constant/constant';
import {connect} from 'react-redux';

const withBigVideo = (Component) => {
  class WithBigVideo extends PureComponent {
    constructor(props) {
      super(props);

      this.videoref = React.createRef();
      this.state = {
        currentTime: 0,
        duration: 0,
        isPlaying: false,
      };

      this._leftTime = this._leftTime.bind(this);
      this._handleIsPlayingChange = this._handleIsPlayingChange.bind(this);
      this._handleOpenFullscreen = this._handleOpenFullscreen.bind(this);
    }

    componentDidMount() {
      const video = this.videoref.current;
      video.src = this.props.film.videoLink;
      video.onloadedmetadata = () =>
        this.setState({
          duration: video.duration,
        });
      video.ontimeupdate = () =>
        this.setState({
          currentTime: Math.trunc(video.currentTime),
        });
    }

    componentDidUpdate() {
      const video = this.videoref.current;
      if (this.state.isPlaying) {
        video.play();
      } else {
        video.pause();
      }
    }

    componentWillUnmount() {
      let video = this.videoref.current;
      video.oncanplaythrough = null;
      video.onplay = null;
      video.onloadedmetadata = null;
      video.ontimeupdate = null;
      video.onpause = null;
      video.controls = null;
      video.src = ``;
    }

    _handleIsPlayingChange() {
      this.setState((prevState) => ({isPlaying: !prevState.isPlaying}));
    }

    _handleOpenFullscreen() {
      const video = this.videoref.current;
      if (video.requestFullscreen) {
        video.requestFullscreen();
      } else if (video.webkitRequestFullscreen) {
        /* Safari */
        video.webkitRequestFullscreen();
      } else if (video.msRequestFullscreen) {
        /* IE11 */
        video.msRequestFullscreen();
      }
    }


    _leftTime() {
      const {currentTime, duration} = this.state;
      const timeDiff = duration - currentTime;
      const second = Math.trunc(timeDiff % 60);
      const minutes = Math.trunc(timeDiff / 60);
      const hours = Math.trunc(minutes / 60);
      return `${hours
        .toString()
        .padStart(2, `0`)}:${minutes
        .toString()
        .padStart(2, `0`)}:${second.toString().padStart(2, `0`)}`;
    }

    render() {
      const {history, filmId, film} = this.props;
      const {backgroundImage, name} = film;

      return (
        <Component
          name={name}
          film={film}
          filmId={filmId}
          history={history}
          leftTime={this._leftTime}
          onOpenFullscreen={this._handleOpenFullscreen}
          isPlaying={this.state.isPlaying}
          duration={this.state.duration}
          currentTime={this.state.currentTime}
          onIsPlayingChange={this._handleIsPlayingChange}
        >
          <video
            ref={this.videoref}
            className="player__video"
            poster={backgroundImage}
          ></video>
        </Component>
      );
    }
  }

  WithBigVideo.propTypes = {
    history: PropTypes.object.isRequired,
    filmId: PropTypes.number.isRequired,
    film: PropTypes.shape({
      name: PropTypes.string.isRequired,
      genre: PropTypes.string.isRequired,
      released: PropTypes.number.isRequired,
      posterImage: PropTypes.string.isRequired,
      previewImage: PropTypes.string.isRequired,
      backgroundImage: PropTypes.string.isRequired,
      backgroundColor: PropTypes.string.isRequired,
      runTime: PropTypes.number.isRequired,
      rating: PropTypes.number.isRequired,
      scoresCount: PropTypes.number.isRequired,
      description: PropTypes.string.isRequired,
      director: PropTypes.string.isRequired,
      starring: PropTypes.arrayOf(
          PropTypes.string.isRequired
      ).isRequired,
      isFavorite: PropTypes.bool.isRequired,
      videoLink: PropTypes.string.isRequired,
      previewVideoLink: PropTypes.string.isRequired,
      id: PropTypes.number.isRequired,
    }).isRequired
  };

  return WithBigVideo;
};

const mapStateToProps = (state, props) => ({
  film: getFilmById(state, props),
});
export {withBigVideo};
export default (Component) => connect(mapStateToProps)(withBigVideo(Component));
