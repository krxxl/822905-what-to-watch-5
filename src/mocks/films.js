export default [{
  name: `The Grand Budapest Hotel`,
  genre: `comedy`,
  year: `2010`,
  poster: `img/the-grand-budapest-hotel-poster.jpg`,
  preview: `img/bohemian-rhapsody.jpg`,
  hero: `img/bg-the-grand-budapest-hotel.jpg`,
  time: `1h 39m`,
  rating: `9.8`,
  count: 240,
  desc: `In the 1930s, the Grand Budapest Hotel is a popular European ski resort, presided over by concierge Gustave H. (Ralph Fiennes). Zero, a junior lobby boy, becomes Gustaves friend and protege.  Gustave prides himself on providing first-class service to the hotels guests, including satisfying the sexual needs of the many elderly women who stay there. When one of Gustaves lovers dies mysteriously, Gustave finds himself the recipient of a priceless painting and the chief suspect in her murder.`,
  director: `Wes Andreson`,
  starring: [`Bill Murray`,
    `Edward Norton`,
    `Jude Law`,
    `Willem Dafoe`,
    `Saoirse Ronan`,
    `Tony Revoloru`],
  video: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
  isInList: true,
  id: 1,
},
{
  name: `bohemian-rhapsody`,
  genre: `Sci-Fi`,
  year: `2010`,
  poster: `img/bohemian-rhapsody.jpg`,
  preview: `img/bohemian-rhapsody.jpg`,
  hero: `img/bg-the-grand-budapest-hotel.jpg`,
  time: `2h 01m`,
  rating: `9.8`,
  count: 240,
  desc: `In the 1930s, the Grand Budapest Hotel is a popular European ski resort, presided over by concierge Gustave H. (Ralph Fiennes). Zero, a junior lobby boy, becomes Gustaves friend and protege.  Gustave prides himself on providing first-class service to the hotels guests, including satisfying the sexual needs of the many elderly women who stay there. When one of Gustaves lovers dies mysteriously, Gustave finds himself the recipient of a priceless painting and the chief suspect in her murder.`,
  director: `Wes Andreson`,
  starring: [`Bill Murray`,
    `Edward Norton`,
    `Jude Law`,
    `Willem Dafoe`,
    `Saoirse Ronan`,
    `Tony Revoloru`],
  video: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
  isInList: true,
  id: 2,
}, {
  name: `dardjeeling-limited`,
  genre: `comedy`,
  year: `2010`,
  poster: `img/dardjeeling-limited.jpg`,
  preview: `img/dardjeeling-limited.jpg`,
  hero: `img/bg-the-grand-budapest-hotel.jpg`,
  time: `1h 01m`,
  rating: `9.8`,
  count: 240,
  desc: `In the 1930s, the Grand Budapest Hotel is a popular European ski resort, presided over by concierge Gustave H. (Ralph Fiennes). Zero, a junior lobby boy, becomes Gustaves friend and protege.  Gustave prides himself on providing first-class service to the hotels guests, including satisfying the sexual needs of the many elderly women who stay there. When one of Gustaves lovers dies mysteriously, Gustave finds himself the recipient of a priceless painting and the chief suspect in her murder.`,
  director: `Wes Andreson`,
  starring: [`Bill Murray`,
    `Edward Norton`,
    `Jude Law`,
    `Willem Dafoe`,
    `Saoirse Ronan`,
    `Tony Revoloru`],
  video: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
  isInList: true,
  id: 3,
}, {
  name: `fantastic-beasts-the-crimes-of-grindelwald`,
  genre: `comedy`,
  year: `2010`,
  poster: `img/fantastic-beasts-the-crimes-of-grindelwald.jpg`,
  preview: `img/fantastic-beasts-the-crimes-of-grindelwald.jpg`,
  hero: `img/bg-the-grand-budapest-hotel.jpg`,
  time: `1h 15m`,
  rating: `9.8`,
  count: 240,
  desc: `In the 1930s, the Grand Budapest Hotel is a popular European ski resort, presided over by concierge Gustave H. (Ralph Fiennes). Zero, a junior lobby boy, becomes Gustaves friend and protege.  Gustave prides himself on providing first-class service to the hotels guests, including satisfying the sexual needs of the many elderly women who stay there. When one of Gustaves lovers dies mysteriously, Gustave finds himself the recipient of a priceless painting and the chief suspect in her murder.`,
  director: `Wes Andreson`,
  starring: [`Bill Murray`,
    `Edward Norton`,
    `Jude Law`,
    `Willem Dafoe`,
    `Saoirse Ronan`,
    `Tony Revoloru`],
  video: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
  isInList: true,
  id: 4,
}, {
  name: `johnny-english`,
  genre: `comedy`,
  year: `2010`,
  poster: `img/johnny-english.jpg`,
  preview: `img/johnny-english.jpg`,
  hero: `img/bg-the-grand-budapest-hotel.jpg`,
  time: `1h 39m`,
  rating: `9.8`,
  count: 240,
  desc: `In the 1930s, the Grand Budapest Hotel is a popular European ski resort, presided over by concierge Gustave H. (Ralph Fiennes). Zero, a junior lobby boy, becomes Gustaves friend and protege.  Gustave prides himself on providing first-class service to the hotels guests, including satisfying the sexual needs of the many elderly women who stay there. When one of Gustaves lovers dies mysteriously, Gustave finds himself the recipient of a priceless painting and the chief suspect in her murder.`,
  director: `Wes Andreson`,
  starring: [`Bill Murray`,
    `Edward Norton`,
    `Jude Law`,
    `Willem Dafoe`,
    `Saoirse Ronan`,
    `Tony Revoloru`],
  video: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
  isInList: true,
  id: 5,
}, {
  name: `macbeth`,
  genre: `comedy`,
  year: `2010`,
  poster: `img/macbeth.jpg`,
  preview: `img/macbeth.jpg`,
  hero: `img/bg-the-grand-budapest-hotel.jpg`,
  time: `1h 39m`,
  rating: `9.8`,
  count: 240,
  desc: `In the 1930s, the Grand Budapest Hotel is a popular European ski resort, presided over by concierge Gustave H. (Ralph Fiennes). Zero, a junior lobby boy, becomes Gustaves friend and protege.  Gustave prides himself on providing first-class service to the hotels guests, including satisfying the sexual needs of the many elderly women who stay there. When one of Gustaves lovers dies mysteriously, Gustave finds himself the recipient of a priceless painting and the chief suspect in her murder.`,
  director: `Wes Andreson`,
  starring: [`Bill Murray`,
    `Edward Norton`,
    `Jude Law`,
    `Willem Dafoe`,
    `Saoirse Ronan`,
    `Tony Revoloru`],
  video: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
  isInList: true,
  id: 6,
}, {
  name: `midnight-special`,
  genre: `kids & family`,
  year: `2010`,
  poster: `img/midnight-special.jpg`,
  preview: `img/midnight-special.jpg`,
  hero: `img/bg-the-grand-budapest-hotel.jpg`,
  time: `1h 39m`,
  rating: `9.8`,
  count: 240,
  desc: `In the 1930s, the Grand Budapest Hotel is a popular European ski resort, presided over by concierge Gustave H. (Ralph Fiennes). Zero, a junior lobby boy, becomes Gustaves friend and protege.  Gustave prides himself on providing first-class service to the hotels guests, including satisfying the sexual needs of the many elderly women who stay there. When one of Gustaves lovers dies mysteriously, Gustave finds himself the recipient of a priceless painting and the chief suspect in her murder.`,
  director: `Wes Andreson`,
  starring: [`Bill Murray`,
    `Edward Norton`,
    `Jude Law`,
    `Willem Dafoe`,
    `Saoirse Ronan`,
    `Tony Revoloru`],
  isInList: true,
  video: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
  id: 7,
}, {
  name: `moonrise-kingdom`,
  genre: `thriller`,
  year: `2010`,
  poster: `img/moonrise-kingdom.jpg`,
  preview: `img/moonrise-kingdom.jpg`,
  hero: `img/bg-the-grand-budapest-hotel.jpg`,
  time: `1h 39m`,
  rating: `9.8`,
  count: 240,
  desc: `In the 1930s, the Grand Budapest Hotel is a popular European ski resort, presided over by concierge Gustave H. (Ralph Fiennes). Zero, a junior lobby boy, becomes Gustaves friend and protege.  Gustave prides himself on providing first-class service to the hotels guests, including satisfying the sexual needs of the many elderly women who stay there. When one of Gustaves lovers dies mysteriously, Gustave finds himself the recipient of a priceless painting and the chief suspect in her murder.`,
  director: `Wes Andreson`,
  starring: [`Bill Murray`,
    `Edward Norton`,
    `Jude Law`,
    `Willem Dafoe`,
    `Saoirse Ronan`,
    `Tony Revoloru`],
  isInList: true,
  video: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
  id: 8,
}, {
  name: `moonrise-kingdom`,
  genre: `horror`,
  year: `2010`,
  poster: `img/moonrise-kingdom.jpg`,
  preview: `img/moonrise-kingdom.jpg`,
  hero: `img/bg-the-grand-budapest-hotel.jpg`,
  time: `1h 39m`,
  rating: `9.8`,
  count: 240,
  desc: `In the 1930s, the Grand Budapest Hotel is a popular European ski resort, presided over by concierge Gustave H. (Ralph Fiennes). Zero, a junior lobby boy, becomes Gustaves friend and protege.  Gustave prides himself on providing first-class service to the hotels guests, including satisfying the sexual needs of the many elderly women who stay there. When one of Gustaves lovers dies mysteriously, Gustave finds himself the recipient of a priceless painting and the chief suspect in her murder.`,
  director: `Wes Andreson`,
  starring: [`Bill Murray`,
    `Edward Norton`,
    `Jude Law`,
    `Willem Dafoe`,
    `Saoirse Ronan`,
    `Tony Revoloru`],
  isInList: true,
  video: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
  id: 9,
}, {
  name: `moonrise-kingdom`,
  genre: `documentary`,
  year: `2010`,
  poster: `img/moonrise-kingdom.jpg`,
  preview: `img/moonrise-kingdom.jpg`,
  hero: `img/bg-the-grand-budapest-hotel.jpg`,
  time: `1h 39m`,
  rating: `9.8`,
  count: 240,
  desc: `In the 1930s, the Grand Budapest Hotel is a popular European ski resort, presided over by concierge Gustave H. (Ralph Fiennes). Zero, a junior lobby boy, becomes Gustaves friend and protege.  Gustave prides himself on providing first-class service to the hotels guests, including satisfying the sexual needs of the many elderly women who stay there. When one of Gustaves lovers dies mysteriously, Gustave finds himself the recipient of a priceless painting and the chief suspect in her murder.`,
  director: `Wes Andreson`,
  starring: [`Bill Murray`,
    `Edward Norton`,
    `Jude Law`,
    `Willem Dafoe`,
    `Saoirse Ronan`,
    `Tony Revoloru`],
  isInList: true,
  video: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
  id: 10,
}, {
  name: `moonrise-kingdom`,
  genre: `drama`,
  year: `2010`,
  poster: `img/moonrise-kingdom.jpg`,
  preview: `img/moonrise-kingdom.jpg`,
  hero: `img/bg-the-grand-budapest-hotel.jpg`,
  time: `1h 39m`,
  rating: `9.8`,
  count: 240,
  desc: `In the 1930s, the Grand Budapest Hotel is a popular European ski resort, presided over by concierge Gustave H. (Ralph Fiennes). Zero, a junior lobby boy, becomes Gustaves friend and protege.  Gustave prides himself on providing first-class service to the hotels guests, including satisfying the sexual needs of the many elderly women who stay there. When one of Gustaves lovers dies mysteriously, Gustave finds himself the recipient of a priceless painting and the chief suspect in her murder.`,
  director: `Wes Andreson`,
  starring: [`Bill Murray`,
    `Edward Norton`,
    `Jude Law`,
    `Willem Dafoe`,
    `Saoirse Ronan`,
    `Tony Revoloru`],
  isInList: true,
  video: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
  id: 11,
}, {
  name: `moonrise-kingdom`,
  genre: `Sci-Fi`,
  year: `2010`,
  poster: `img/moonrise-kingdom.jpg`,
  preview: `img/moonrise-kingdom.jpg`,
  hero: `img/bg-the-grand-budapest-hotel.jpg`,
  time: `1h 39m`,
  rating: `9.8`,
  count: 240,
  desc: `In the 1930s, the Grand Budapest Hotel is a popular European ski resort, presided over by concierge Gustave H. (Ralph Fiennes). Zero, a junior lobby boy, becomes Gustaves friend and protege.  Gustave prides himself on providing first-class service to the hotels guests, including satisfying the sexual needs of the many elderly women who stay there. When one of Gustaves lovers dies mysteriously, Gustave finds himself the recipient of a priceless painting and the chief suspect in her murder.`,
  director: `Wes Andreson`,
  starring: [`Bill Murray`,
    `Edward Norton`,
    `Jude Law`,
    `Willem Dafoe`,
    `Saoirse Ronan`,
    `Tony Revoloru`],
  isInList: true,
  video: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
  id: 12,
}, {
  name: `moonrise-kingdom`,
  genre: `Romance`,
  year: `2010`,
  poster: `img/moonrise-kingdom.jpg`,
  preview: `img/moonrise-kingdom.jpg`,
  hero: `img/bg-the-grand-budapest-hotel.jpg`,
  time: `1h 39m`,
  rating: `9.8`,
  count: 240,
  desc: `In the 1930s, the Grand Budapest Hotel is a popular European ski resort, presided over by concierge Gustave H. (Ralph Fiennes). Zero, a junior lobby boy, becomes Gustaves friend and protege.  Gustave prides himself on providing first-class service to the hotels guests, including satisfying the sexual needs of the many elderly women who stay there. When one of Gustaves lovers dies mysteriously, Gustave finds himself the recipient of a priceless painting and the chief suspect in her murder.`,
  director: `Wes Andreson`,
  starring: [`Bill Murray`,
    `Edward Norton`,
    `Jude Law`,
    `Willem Dafoe`,
    `Saoirse Ronan`,
    `Tony Revoloru`],
  isInList: true,
  video: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
  id: 13,
}, {
  name: `moonrise-kingdom`,
  genre: `Kids & Family`,
  year: `2010`,
  poster: `img/moonrise-kingdom.jpg`,
  preview: `img/moonrise-kingdom.jpg`,
  hero: `img/bg-the-grand-budapest-hotel.jpg`,
  time: `1h 39m`,
  rating: `9.8`,
  count: 240,
  desc: `In the 1930s, the Grand Budapest Hotel is a popular European ski resort, presided over by concierge Gustave H. (Ralph Fiennes). Zero, a junior lobby boy, becomes Gustaves friend and protege.  Gustave prides himself on providing first-class service to the hotels guests, including satisfying the sexual needs of the many elderly women who stay there. When one of Gustaves lovers dies mysteriously, Gustave finds himself the recipient of a priceless painting and the chief suspect in her murder.`,
  director: `Wes Andreson`,
  starring: [`Bill Murray`,
    `Edward Norton`,
    `Jude Law`,
    `Willem Dafoe`,
    `Saoirse Ronan`,
    `Tony Revoloru`],
  isInList: true,
  video: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
  id: 14,
}, {
  name: `moonrise-kingdom`,
  genre: `Horror`,
  year: `2010`,
  poster: `img/moonrise-kingdom.jpg`,
  preview: `img/moonrise-kingdom.jpg`,
  hero: `img/bg-the-grand-budapest-hotel.jpg`,
  time: `1h 39m`,
  rating: `9.8`,
  count: 240,
  desc: `In the 1930s, the Grand Budapest Hotel is a popular European ski resort, presided over by concierge Gustave H. (Ralph Fiennes). Zero, a junior lobby boy, becomes Gustaves friend and protege.  Gustave prides himself on providing first-class service to the hotels guests, including satisfying the sexual needs of the many elderly women who stay there. When one of Gustaves lovers dies mysteriously, Gustave finds himself the recipient of a priceless painting and the chief suspect in her murder.`,
  director: `Wes Andreson`,
  starring: [`Bill Murray`,
    `Edward Norton`,
    `Jude Law`,
    `Willem Dafoe`,
    `Saoirse Ronan`,
    `Tony Revoloru`],
  isInList: true,
  video: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
  id: 15,
}, {
  name: `moonrise-kingdom`,
  genre: `Drama`,
  year: `2010`,
  poster: `img/moonrise-kingdom.jpg`,
  preview: `img/moonrise-kingdom.jpg`,
  hero: `img/bg-the-grand-budapest-hotel.jpg`,
  time: `1h 39m`,
  rating: `9.8`,
  count: 240,
  desc: `In the 1930s, the Grand Budapest Hotel is a popular European ski resort, presided over by concierge Gustave H. (Ralph Fiennes). Zero, a junior lobby boy, becomes Gustaves friend and protege.  Gustave prides himself on providing first-class service to the hotels guests, including satisfying the sexual needs of the many elderly women who stay there. When one of Gustaves lovers dies mysteriously, Gustave finds himself the recipient of a priceless painting and the chief suspect in her murder.`,
  director: `Wes Andreson`,
  starring: [`Bill Murray`,
    `Edward Norton`,
    `Jude Law`,
    `Willem Dafoe`,
    `Saoirse Ronan`,
    `Tony Revoloru`],
  isInList: true,
  video: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
  id: 16,
}, {
  name: `moonrise-kingdom`,
  genre: `Documentary`,
  year: `2010`,
  poster: `img/moonrise-kingdom.jpg`,
  preview: `img/moonrise-kingdom.jpg`,
  hero: `img/bg-the-grand-budapest-hotel.jpg`,
  time: `1h 39m`,
  rating: `9.8`,
  count: 240,
  desc: `In the 1930s, the Grand Budapest Hotel is a popular European ski resort, presided over by concierge Gustave H. (Ralph Fiennes). Zero, a junior lobby boy, becomes Gustaves friend and protege.  Gustave prides himself on providing first-class service to the hotels guests, including satisfying the sexual needs of the many elderly women who stay there. When one of Gustaves lovers dies mysteriously, Gustave finds himself the recipient of a priceless painting and the chief suspect in her murder.`,
  director: `Wes Andreson`,
  starring: [`Bill Murray`,
    `Edward Norton`,
    `Jude Law`,
    `Willem Dafoe`,
    `Saoirse Ronan`,
    `Tony Revoloru`],
  isInList: true,
  video: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
  id: 17,
}, {
  name: `moonrise-kingdom`,
  genre: `Crime`,
  year: `2010`,
  poster: `img/moonrise-kingdom.jpg`,
  preview: `img/moonrise-kingdom.jpg`,
  hero: `img/bg-the-grand-budapest-hotel.jpg`,
  time: `1h 39m`,
  rating: `9.8`,
  count: 240,
  desc: `In the 1930s, the Grand Budapest Hotel is a popular European ski resort, presided over by concierge Gustave H. (Ralph Fiennes). Zero, a junior lobby boy, becomes Gustaves friend and protege.  Gustave prides himself on providing first-class service to the hotels guests, including satisfying the sexual needs of the many elderly women who stay there. When one of Gustaves lovers dies mysteriously, Gustave finds himself the recipient of a priceless painting and the chief suspect in her murder.`,
  director: `Wes Andreson`,
  starring: [`Bill Murray`,
    `Edward Norton`,
    `Jude Law`,
    `Willem Dafoe`,
    `Saoirse Ronan`,
    `Tony Revoloru`],
  isInList: true,
  video: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
  id: 18,
}, {
  name: `dardjeeling-limited`,
  genre: `comedy`,
  year: `2010`,
  poster: `img/dardjeeling-limited.jpg`,
  preview: `img/dardjeeling-limited.jpg`,
  hero: `img/bg-the-grand-budapest-hotel.jpg`,
  time: `1h 01m`,
  rating: `9.8`,
  count: 240,
  desc: `In the 1930s, the Grand Budapest Hotel is a popular European ski resort, presided over by concierge Gustave H. (Ralph Fiennes). Zero, a junior lobby boy, becomes Gustaves friend and protege.  Gustave prides himself on providing first-class service to the hotels guests, including satisfying the sexual needs of the many elderly women who stay there. When one of Gustaves lovers dies mysteriously, Gustave finds himself the recipient of a priceless painting and the chief suspect in her murder.`,
  director: `Wes Andreson`,
  starring: [`Bill Murray`,
    `Edward Norton`,
    `Jude Law`,
    `Willem Dafoe`,
    `Saoirse Ronan`,
    `Tony Revoloru`],
  video: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
  isInList: true,
  id: 19,
}, {
  name: `dardjeeling-limited`,
  genre: `comedy`,
  year: `2010`,
  poster: `img/dardjeeling-limited.jpg`,
  preview: `img/dardjeeling-limited.jpg`,
  hero: `img/bg-the-grand-budapest-hotel.jpg`,
  time: `1h 01m`,
  rating: `9.8`,
  count: 240,
  desc: `In the 1930s, the Grand Budapest Hotel is a popular European ski resort, presided over by concierge Gustave H. (Ralph Fiennes). Zero, a junior lobby boy, becomes Gustaves friend and protege.  Gustave prides himself on providing first-class service to the hotels guests, including satisfying the sexual needs of the many elderly women who stay there. When one of Gustaves lovers dies mysteriously, Gustave finds himself the recipient of a priceless painting and the chief suspect in her murder.`,
  director: `Wes Andreson`,
  starring: [`Bill Murray`,
    `Edward Norton`,
    `Jude Law`,
    `Willem Dafoe`,
    `Saoirse Ronan`,
    `Tony Revoloru`],
  video: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
  isInList: true,
  id: 20,
},
];
