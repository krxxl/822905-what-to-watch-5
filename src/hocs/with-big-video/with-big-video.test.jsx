import React from 'react';
import renderer from 'react-test-renderer';
import {withBigVideo} from './with-big-video';
import PropTypes from "prop-types";
import films from '../../mocks/films';

const mockComponent = (props) => {
  const {children} = props;

  return <div>
    {children}
  </div>;
};

mockComponent.propTypes = {
  children: PropTypes.node.isRequired,
};

const MockComponentWrapped = withBigVideo(mockComponent);

describe(`Render withBigVideo`, () => {
  it(`Render withBigVideo`, () => {
    const tree = renderer
      .create(
          <MockComponentWrapped
            film={films[0]}
            filmId={films[0].id}
            history={{}}
            isFavorite={false}
            videoLink={films[0].videoLink}
            previewVideoLink={films[0].previewVideoLink}
            id={films[0].id}
          />
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
