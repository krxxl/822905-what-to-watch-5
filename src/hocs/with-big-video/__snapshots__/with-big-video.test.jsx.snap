// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render withBigVideo Render withBigVideo 1`] = `
<div>
  <video
    className="player__video"
    poster="img/bg-the-grand-budapest-hotel.jpg"
  />
</div>
`;
